datalad/support/gitrepo.py

```diff
@@ -2661,16 +2718,15 @@ def get_tags(self, output=None):
           is attached to. The list is sorted by commit date, with the most
           recent commit being the last element.
         """
-        tag_objs = sorted(
-            self.repo.tags,
-            key=lambda t: t.commit.committed_date
-        )
         tags = [
-            {
-                'name': t.name,
-                'hexsha': t.commit.hexsha
-             }
-            for t in tag_objs
+            dict(
+                name=t['refname:lstrip=2'],
+                hexsha=t['object'] if t['object'] else t['objectname'],
+            )
+            for t in self.for_each_ref_(
+                fields=['refname:lstrip=2', 'objectname', 'object'],
```

Comment:
Ah, neat, I didn't realize fields for `object` (and other header values) existed.  So this is equivalent to dereferencing `objectname`: `['refname:lstrip=2', 'objectname', '*objectname']`.

For sorting, `taggerdate` isn't ideal because it only applies to annotated tags.  `creatordate` is better here.

I was a bit surprised to see you use ":lstrip" here rather than splicing (so `[10:]`) like you do elsewhere.  I think either is fine, though if we're going to use one consistently, I have a weak preference for using ":lstrip", or actually ":strip".  ":strip" has been around since Git v2.7.1, while the synonym ":lstrip" has only been around since v2.13.0.  While I personally don't think we should worry about supporting v2.13.0, we might as well use the older spelling.  (We really should specify a minimum Git version so that we can make clear decisions in these cases.)

My other related thought was whether you should use `--short` for symbolic-ref and `:short` for for-each-ref to let git handle the truncation, keeping what is needed for the ref to be unambiguous (e.g., refs/heads/foo -> heads/foo and refs/tags/foo -> tags/foo).  But on second thought, there are at least two reasons not to qualify ambiguous refs: (1) the gitpy versions you are replacing do not and (2) get_active_branch(), get_{remote_,}branches}(), and get_tags() are by definition limited to a particular namespace, so we do not need to qualify that the ref is ambiguous in some other namespace.

Reply:
Ha, I didn't know that you can dereference objectname! ;-)

I now uniformly do `:strip=2`. Thx for illustrating the problem space.

I agree re `--short`, and left it as-is.